src/components/LoginModal/loginModalReducer.ts

```diff
@@ -0,0 +1,25 @@
+import produce, { Draft } from 'immer'
+import { HIDE_LOGIN_MODAL, SHOW_LOGIN_MODAL } from './loginModalActionTypes'
+import { LoginModalState } from './loginModalSelector'
+
+const initialState = {
+    isDisplayed: false,
+}
+
+export const loginModalReducer = produce(
+    (draft: Draft<LoginModalState>, { type, payload }) => {
+        switch (type) {
+            case HIDE_LOGIN_MODAL:
+                return {
+                    ...draft,
+                    isDisplayed: false,
+                }
+            case SHOW_LOGIN_MODAL:
+                return {
+                    ...draft,
+                    isDisplayed: true,
```

Comment:
if you only mutate one field, you can do draft.isDisplayed directly. 
This where needed for destructuring the whole object in draft, which is not supported by immer.